src/pages/sign-in/index.tsx

```diff
@@ -9,15 +9,31 @@ import { PeerPrepRectLogo } from "~/assets/logo";
 import LoginWithCredentials from "~/components/Login";
 import Link from "next/link";
 import toast from "react-hot-toast";
+import { useState } from "react";
```

Comment:
Something additional, don't have to add in this pr, but shufan mentioned that we need to display error when we redirect, so I was thinking we can handle that in the either in the with auth wrapper (by rendering an actual component instead of loading on unauthorised), or we can pass a prop into the sign in page on redirect, which should be supported by useSession.

Reply:
ooh thats what i tried for this issue as well: rendering a toast on the new page (it was tough cos this home page had to reload)

But for redirection to signin should be manageable thanks for suggestion

Reply:
https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g39/issues/68
added an issue will do it in a separate PR